### PR TITLE
Shipping fields

### PR DESCRIPTION
**Pull request by sktoiva**

### CODE REVIEW

File: app/models/listing.rb [hunk 1]
```diff
@@ -406,4 +409,10 @@ def answer_for(custom_field)
   def payment_required_at?(community)
     transaction_type.price_field? && community.payments_in_use?
   end
+
+  def currency=(currency)
+    raise ArgumentError.new "Invalid currency. Tried to change #{self[:currency]} to #{currency}." if self[:currency] && self[:currency] != currency
```
Comment (rap1ds) [inline app/models/listing.rb]: What does this do? Did I understand right, it throws when ever someone is trying to change the currency?

Comment (sktoiva) [inline app/models/listing.rb]: Yes. It does just that.

File: db/migrate/20150212124533_add_shipping_enabled_to_community.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddShippingEnabledToCommunity < ActiveRecord::Migration
+  def change
+    add_column :communities, :shipping_enabled, :integer, default: 0
```
Comment (rap1ds) [inline db/migrate/20150212124533_add_shipping_enabled_to_community.rb]: Is this meant to be integer, or should it be boolean?

File: app/models/listing.rb (hunk 1 above)
Comment (rap1ds) [inline app/models/listing.rb]: Could you elaborate why is this needed?

File: db/migrate/20150212124533_add_shipping_enabled_to_community.rb (hunk 2 above)
Comment (sktoiva) [inline db/migrate/20150212124533_add_shipping_enabled_to_community.rb]: Yeah, should be boolean.

Comment (ovan) [inline db/migrate/20150212124533_add_shipping_enabled_to_community.rb]: I'm not sure if it's good idea to put this into community table even temporarily. I spent considerable amount of time moving the paypal enabled out of community table after the first release when the check had been embedded already into a lot of places in the code base. What about marketplace_settings or listing_settings or selling_settings or something like that, per community table?

File: db/migrate/20150212125111_add_shipping_fields_to_listing.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class AddShippingFieldsToListing < ActiveRecord::Migration
+  def change
+    add_column :listings, :require_shipping_address, :boolean, default: false
+    add_column :listings, :pickup_enabled, :boolean, default: true
```
Comment (Gnito) [inline db/migrate/20150212125111_add_shipping_fields_to_listing.rb]: Is true good for default (sounds like it, but might need some discussion)

**Comment by Gnito** [comment]
Update export script

**Comment by Gnito** [comment]
Those are the only things that came to mind.
so, :+1:
